tests/e2e/test_pvc_creation_scale.py

```diff
@@ -0,0 +1,164 @@
+        for k, v in pvc_create_time.items():
+            csv_obj.writerow([k, v])
+
+    @polarion_id('OCS-1885')
+    @pytest.mark.usefixtures(namespace.__name__)
+    def test_all_4_type_pvc_creation_scale(self, namespace, teardown_factory):
+        """
+        Measuring PVC creation time while scaling PVC of all 4 types, Total 1500 PVCs
+        will be created, i.e. 375 each pvc type
+        """
+        number_of_pvc = 375
+        log.info(f"Start creating {number_of_pvc} PVC of all 4 types")
+
+        cephfs_sc_obj = constants.DEFAULT_STORAGECLASS_CEPHFS
+        rbd_sc_obj = constants.DEFAULT_STORAGECLASS_RBD
+
+        fs_pvc_obj, rbd_pvc_obj = ([] for i in range(2))
+        for mode in [constants.ACCESS_MODE_RWO, constants.ACCESS_MODE_RWX]:
+            fs_pvc_obj.extend(helpers.create_multiple_pvcs(
+                sc_name=cephfs_sc_obj, namespace=self.namespace, number_of_pvc=number_of_pvc,
+                size=f"{random.randrange(5, 105, 5)}Gi", access_mode=mode)
+            )
+            rbd_pvc_obj.extend(helpers.create_multiple_pvcs(
+                sc_name=rbd_sc_obj, namespace=self.namespace, number_of_pvc=number_of_pvc,
+                size=f"{random.randrange(5, 105, 5)}Gi", access_mode=mode)
+            )
+
+        fs_pvc_name, rbd_pvc_name = ([] for i in range(2))
+        for fs_obj, rbd_obj in zip(fs_pvc_obj, rbd_pvc_obj):
+            fs_pvc_name.append(fs_obj.name)
+            rbd_pvc_name.append(rbd_obj.name)
+
+        # Check for PVC status using threads
+        threads = list()
+        for obj in fs_pvc_obj:
+            process = threading.Thread(
+                target=helpers.wait_for_resource_state,
+                args=(obj, constants.STATUS_BOUND, )
+            )
+            process.start()
+            threads.append(process)
+        for obj in rbd_pvc_obj:
+            process = threading.Thread(
+                target=helpers.wait_for_resource_state,
+                args=(obj, constants.STATUS_BOUND,)
+            )
+            process.start()
+            threads.append(process)
+        for process in threads:
+            process.join()
+
+        fs_pvc_create_time = helpers.measure_pvc_creation_time_bulk(
+            interface=constants.CEPHFS_INTERFACE, pvc_name_list=fs_pvc_name
+        )
+        rbd_pvc_create_time = helpers.measure_pvc_creation_time_bulk(
+            interface=constants.CEPHBLOCKPOOL, pvc_name_list=rbd_pvc_name
+        )
+        fs_pvc_create_time.update(rbd_pvc_create_time)
+
+        # TODO: Update below code with google API, to record value in spreadsheet
+        # TODO: For now observing Google API limit to write more than 100 writes
+        csv_obj = csv.writer(open("/tmp/All-type-PVC-Creation-Scale.csv", "w"))
+        for k, v in fs_pvc_create_time.items():
+            csv_obj.writerow([k, v])
+
+        for fs_obj, rbd_obj in zip(fs_pvc_obj, rbd_pvc_obj):
+            teardown_factory(fs_obj)
+            teardown_factory(rbd_obj)
```

Comment:
can this be added to threads which will make deletion faster instead of deleting one by one serially

Reply:
Planning to implement capturing deletion time, so it will be sequential but will be removing teardown_factory